File: src/main/java/com/geminibot/geminibot/entities/postgres/ApiKey.java
package com.geminibot.geminibot.entities.postgres;

import com.geminibot.geminibot.entities.constants.ApiKeyType;

import javax.persistence.*;

@Entity
@Table(name="api_keys")
public class ApiKey {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;

    @Column(unique=true)
    private ApiKeyType type;
    private String publicKey;
    private String secretKey;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public ApiKeyType getType() {
        return type;
    }

    public void setType(ApiKeyType type) {
        this.type = type;
    }

    public String getPublicKey() {
        return publicKey;
    }

    public void setPublicKey(String publicKey) {
        this.publicKey = publicKey;
    }

    public String getSecretKey() {
        return secretKey;
    }

    public void setSecretKey(String secretKey) {
        this.secretKey = secretKey;
    }
}

File: react-app/src/schemas/ApiKey.ts
import * as yup from "yup";
import { SchemaOf } from "yup";

interface ApiKey {
  type: string;
  publicKey: string;
  secretKey: string;
}

const ApiKeyValidationSchema: SchemaOf<ApiKey> = yup
  .object({
    type: yup.string().required(),
    publicKey: yup.string().required(),
    secretKey: yup.string().required(),
  })
  .required();

export { ApiKeyValidationSchema };
export type { ApiKey };
